coalib/settings/Section.py

```diff
@@ -43,6 +45,52 @@ def append_to_sections(sections,
         key, str(value), origin, from_cli=from_cli, to_append=to_append))
 
 
+def extract_aspects_from_section(section):
+    """
+    Extracts aspects and their related settings from a section and create an
+    aspectlist from it.
+
+    :param section: Section object.
+    :return:        aspectlist containing aspectclass instance with
+                    user-defined tastes.
+    """
+    # Skip aspects initialization if not configured in section
+    if not len(section.get('aspects')):
+        return None
+
+    if not len(section.get('language')):
+        raise AttributeError('Language was not found in configuration file. '
+                             'Usage of aspect-based configuration must include'
+                             'language information.')
+        # TODO acquire language from interactive input instead of just raising
```

Comment:
Noooo ... never mix parsing/extraction and UI.  extraction methods should fail if they cant extract, and let the caller do any UI *if* they want a fallback.